x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -144,6 +144,15 @@ ASTERISK : '*';
 SLASH : '/';
 PERCENT : '%';
 
+NAMED_PARAM
+    : '?' LETTER UNQUOTED_ID_BODY*
+    | '?' (UNDERSCORE | ASPERAND) UNQUOTED_ID_BODY+
+    ;
+
+POSITIONAL_PARAM
+    : '?' DIGIT+
+    ;
```

Comment:
According to the discussion with Costin, we will combine NAMED_PARAM and POSITIONAL_PARAM to one token and let the rule to figure them out, so that we can have more flexibility/restriction/control on the name of the parameter, and a better error message if needed.